catalog/ui/search-ui/standard/src/main/webapp/js/controllers/cesium.controller.js

```diff
@@ -59,10 +62,25 @@ define(['application',
                 this.listenTo(wreqr.vent, 'metacard:overlay', this.overlayImage);
                 this.listenTo(wreqr.vent, 'metacard:overlay:remove', this.removeOverlay);
 
+                this.clustering = false;
+
+                $(".cesium-viewer-toolbar").append("<button class='cesium-button cesium-toolbar-button cesium-home-button'><span class='cluster-results'></span></button>");
+
                 if (wreqr.reqres.hasHandler('search:results')) {
                     this.newResults(wreqr.reqres.request('search:results'));
                 }
             },
+
+            toggleClustering: function() {
+                if(!this.clustering) {
+                    this.clustering = true;
+                    mapclustering.clusteringAlgorithm();
```

Comment:
Can you change `clusteringAlgorithm` and `resetBillboard` to `cluster` and `uncluster`?  Algorithm in a method name seems odd to me since it makes it a noun instead of an action and reset billboard is leaking implementation details.